src/index.js

```diff
@@ -158,7 +158,7 @@ function createLoadableComponent(loadFn, options) {
       return init();
     }
 
-    componentWillMount() {
+    componentDidMount() {
```

Comment:
This could introduce bugs, why don't you go the safe route and use UNSAFE_componentWillMount for now?

If you really want to go directly to didMount, `this._mounted = true` would need to move it to the constructor.
Calling `retry` from the loading component before the mount phase could cause unexpected behaviors with this change, for example.
The `_loadModule` is probably fine to stay here.

Unfortunately sounds like nobody will ever merge this anyway...